kernel/core/os.adb

```diff
@@ -956,7 +956,7 @@ is
                   --  in other case, for now we "compact" the rest of the mbx
                   --  queue, so that there is no "hole" in it for the next mbx
                   --  search.
-                  for iterator2 in iterator .. os_mbx_get_mbx_count (current)
+                  for iterator2 in os_mbx_count_t'Succ (iterator) .. os_mbx_count_t'Pred (os_mbx_get_mbx_count (current))
```

Comment:
As iterator is of type os_mbx_count_t (which is not modular) it might raise exception if it exceed 'Last with the 'Succ operation. This is something that should be avoided in Spark.